python/pyarrow/pandas_compat.py

```diff
@@ -544,9 +547,14 @@ def table_to_blockmanager(options, table, memory_pool, nthreads=1,
 
     column_strings = [x.name for x in block_table.itercolumns()]
     if columns:
-        columns_name_dict = {
-            c.get('field_name', str(c['name'])): c['name'] for c in columns
-        }
+        columns_name_dict = {}
+        for c in columns:
+            column_name = c['name']
+            if not isinstance(column_name, six.text_type):
+                column_name = frombytes(str(column_name))
```

Comment:
`frombytes` works only on `bytes`. Thus the above code is valid in Python 2 but breaks the unittests for Python 3. Removing the `str` should fix this.